app/src/androidTest/java/org/mozilla/focus/activity/ImageSelectTest.java

```diff
@@ -0,0 +1,138 @@
+        @Override
+        protected void beforeActivityLaunched() {
+            super.beforeActivityLaunched();
+
+            appContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext()
+                    .getApplicationContext();
+
+            PreferenceManager.getDefaultSharedPreferences(appContext)
+                    .edit()
+                    .putBoolean(FIRSTRUN_PREF, true)
+                    .apply();
+
+            webServer = new MockWebServer();
+
+            try {
+                webServer.enqueue(new MockResponse()
+                        .setBody(readTestAsset("test.html")));
+                webServer.enqueue(new MockResponse()
+                        .setBody(readTestAsset("rabbit.jpg"))
+                        .setHeader("Content-Type", "image/jpg"));
+                webServer.enqueue(new MockResponse()
+                        .setBody(readTestAsset("service-worker.js"))
+                        .setHeader("Content-Type", "text/javascript"));
+                webServer.start();
+            } catch (IOException e) {
+                throw new AssertionError("Could not start web server", e);
+            }
+        }
+
+        @Override
+        protected void afterActivityFinished() {
+            super.afterActivityFinished();
+
+            try {
+                webServer.shutdown();
+            } catch (IOException e) {
+                throw new AssertionError("Could not stop web server", e);
+            }
+        }
+    };
+
+    @Test
+    public void DeleteWebViewDataTest() throws InterruptedException, UiObjectNotFoundException, IOException {
+        final long waitingTime = TestHelper.waitingTime;
+
+        // Load website with service worker
+
+        TestHelper.urlBar.waitForExists(waitingTime);
+        TestHelper.urlBar.click();
+        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+        TestHelper.inlineAutocompleteEditText.clearTextField();
+        TestHelper.inlineAutocompleteEditText.setText(webServer.url(TEST_PATH).toString());
+        TestHelper.hint.waitForExists(waitingTime);
+        TestHelper.pressEnterKey();
+        TestHelper.webView.waitForExists(waitingTime);
+
+        // Assert website is loaded
+
+        final UiObject titleMsg = TestHelper.mDevice.findObject(new UiSelector()
+                .description("focus test page")
+                .enabled(true));
+        titleMsg.waitForExists(waitingTime);
+        assertTrue("Website title loaded", titleMsg.exists());
+
+        // Erase browsing session
+        TestHelper.floatingEraseButton.perform(click());
+        TestHelper.erasedMsg.waitForExists(waitingTime);
+        Assert.assertTrue(TestHelper.erasedMsg.exists());
+        Assert.assertTrue(TestHelper.urlBar.exists());
+        TestHelper.waitForIdle();
+    }
+
+    private String readTestAsset(String filename) throws IOException {
```

Comment:
Is this intentionally duplicated (readTestAsset/readStreamIntoString) - can we just use the TestHelper version instead (they'd need to be made public I guess)?